proj1a/ArrayDeque.java

```diff
@@ -0,0 +1,203 @@
+public class ArrayDeque<T> implements Deque<T> {
+    private T[] items;
+    private int size;
+    private int nextFirst;
+    private int nextLast;
+    private int REFACTOR = 2; //resize factor
+
+    /**
+     * create empty array deque
+     */
+    public ArrayDeque() {
+        items = (T[]) new Object[8];
+        size = 0;
+        nextFirst = 4;
+        nextLast = 5;
+    }
+
+    /**
+     * calculate nextFirst
+     * @param i: current position
+     */
+    private int nextF(int i) {
+        return i == 0? items.length - 1 : i -1;
+    }
+
+    /**
+     * calculate nextLast
+     * @param i: current position
+     */
+    private int nextL(int i) {
+        /*
+        int next;
+        if (i == items.length - 1) {
+            next = 0;
+        } else {
+            next = i + 1;
+        }*/
+        return i == items.length - 1? 0 : i + 1;
+    }
+
+    /**
+     * resize array, create a larger one
+     * @param capacity: new size for the array
+     */
+    private void resize(int capacity) {
+        T[] temp = (T[]) new Object[capacity];
+        int old_size = size;
+        int first = nextF(nextFirst);
+
+        for (int i = 0; i < old_size ; i++ ) {
+            temp[i] = items[first];
+            first = nextF(first);
+        }
+        nextFirst = capacity - 1;
+        nextLast = size;
+        items = temp;
+    }
+
+    /**
+     * add item to the front of the deque
+     * @param item: item to add
+     */
+    @Override
+    public void addFirst(T item) {
```

Comment:
Add `synchronized` to make it thread-safe

Reply:
``` synchorized``` added